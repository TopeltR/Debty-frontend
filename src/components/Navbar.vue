<template>
    <b-container fluid>
        <b-row class='PB65px'>
            <b-navbar class='fixed-top' style='background-color: #E8E8E8; width:100%' type='light' variant='' toggleable='sm'>
                <b-navbar-toggle target='nav_collapse'></b-navbar-toggle>
                <b-navbar-brand class='limegreen brand' v-on:click='goToHome'>debty</b-navbar-brand>
                <b-collapse is-nav id='nav_collapse'>
                    <b-navbar-nav>
                        <b-nav-item to='/home'>
                            Home
                        </b-nav-item>
                    </b-navbar-nav>
                    <b-navbar-nav>
                        <b-nav-item to='/events'>
                            Events
                        </b-nav-item>
                    </b-navbar-nav>
                    <b-navbar-nav>
                        <b-nav-item to='/debts'>
                            Debts
                        </b-nav-item>
                    </b-navbar-nav>
                </b-collapse>
            </b-navbar>
        </b-row>
    </b-container>
</template>

<script>
    import router from '@/router';

    export default {
        name: 'Navbar',
        methods: {
            goToHome() {
                router.push('/home');
            },
        }
    }
</script>

<style scoped>
    .limegreen {
        color: limegreen;
    }

    .brand {
        font-weight: bolder;
        font-size: 30px;
    }
    .PB65px {
        padding-bottom: 65px;
    }
</style>