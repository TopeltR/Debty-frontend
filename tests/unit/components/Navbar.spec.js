import Navbar from "@/components/Navbar";
import userStore from '@/stores/UserStore'
import {mount} from "@vue/test-utils";
import Vue from "vue";
import BootstrapVue from 'bootstrap-vue';

describe('Navbar', () => {
    Vue.use(BootstrapVue);
    console.error = jest.fn();

    const get = jest.fn(() => Promise.resolve({data: [1, 2, 3]}));
    Vue.prototype.$http = {get};
    userStore.getUser = jest.fn(() => Promise.resolve({id: 1}));

    describe('getNotificationCount', () => {
        it('should request friend requests from /contact/waiting/{user.id} and set notificationAmount to length of returned array', async () => {
            const navbar = mount(Navbar);
            await Vue.nextTick();

            await navbar.vm.getNotificationCount();

            expect(get).toHaveBeenCalledWith('/contact/waiting/1');
            expect(navbar.vm.notification).toBe(true);
            expect(navbar.vm.notificationAmount).toBe(3);
        })
    });

    it('should render', () => {
        const navbar = mount(Navbar);

        expect(navbar.html()).toMatchSnapshot();
    })
});
