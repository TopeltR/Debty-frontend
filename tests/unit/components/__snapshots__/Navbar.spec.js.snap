// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render 1`] = `
<div class="container-fluid">
  <div class="row">
    <nav class="navbar fixed-top w-100 bg-light-gray navbar-light navbar-expand-sm"><button type="button" aria-label="Toggle navigation" aria-controls="nav_collapse" aria-expanded="false" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
      <div class="navbar-brand limegreen brand">debty</div>
      <div id="nav_collapse" class="navbar-collapse collapse" style="display: none;">
        <ul class="navbar-nav">
          <li class="nav-item"><a target="_self" href="/home" class="nav-link">
              Home
            </a></li>
        </ul>
        <ul class="navbar-nav">
          <li class="nav-item"><a target="_self" href="/events" class="nav-link">
              Events
            </a></li>
        </ul>
        <ul class="navbar-nav">
          <li class="nav-item"><a target="_self" href="/debts" class="nav-link">
              Debts
            </a></li>
        </ul>
        <ul class="navbar-nav">
          <li class="nav-item"><a target="_self" href="/contacts" class="nav-link">
              Contacts
              <!----></a></li>
        </ul>
        <ul class="navbar-nav">
          <li class="nav-item"><a target="_self" href="/profile" class="nav-link">
              Profile
            </a></li>
        </ul>
        <ul class="navbar-nav ml-auto"><span class="d-none d-md-block mr-4 mt-2"> Welcome, <b></b></span>
          <li class="nav-item"><a target="_self" href="#" class="nav-link">
              <font-awesome-icon icon="sign-out-alt" class="icon"></font-awesome-icon> <span>Sign out</span>
            </a></li>
        </ul>
      </div>
    </nav>
  </div>
</div>
`;
